src/db.ts

```diff
@@ -34,58 +33,49 @@ export class DB {
   /**
    * Retrieves a raw value from leveldb.
    * @param {Buffer} key
-   * @param {Function} cb A callback `Function`, which is given the arguments
-   * `err` - for errors that may have occured
-   * and `value` - the found value in a `Buffer` or if no value was found `null`.
+   * @returns {Promise} - Promise resolves with `Buffer` if a value is found or `null` if no value is found.
    */
-  get(key: Buffer, cb: BufferCallback) {
-    if (!Buffer.isBuffer(key)) throw new Error('Invalid input: expected buffer')
-
-    this._leveldb.get(key, ENCODING_OPTS, (err?: Error, v?: Buffer) => {
-      if (err || !v) {
-        cb(null, null)
+  async get(key: Buffer): Promise<Buffer | null> {
+    let value = null
+    try {
+      value = await this._leveldb.get(key, ENCODING_OPTS)
+    } catch (error) {
+      if (error.notFound) {
+        console.log('not found')
+        // throw error
```

Comment:
What is the best way to do here? Return `null`? or throw an error? //cc @s1na

Reply:
Leveldb was throwing a ErrorNotFound so i was playing around in a few places where I’d catch it and return null (I believe that is the correct behavior when getting an unknown key, should return null). In the current code I believe the error calls back to `onDone()`.

I still have a few other things left to resolve but can let you know as soon as it’s ready for some in depth review :)

Thanks!

Reply:
Yah, I am just strolling a bit over out of interest! 😄